src/Zope2/Startup/tests/test_schema.py

```diff
@@ -190,3 +190,46 @@ def test_ms_public_header(self):
             self.assertFalse(webdav.enable_ms_public_header)
         finally:
             webdav.enable_ms_public_header = default_setting
+
+    def test_dos_protection(self):
+        from ZPublisher import HTTPRequest
+
+        params = ["FORM_%s_LIMIT" % name
+                  for name in ("MEMORY", "DISK", "MEMFILE")]
+        defaults = dict((name, getattr(HTTPRequest, name)) for name in params)
```

Comment:
Why not using the more modern approach?
```suggestion
        defaults = {name: getattr(HTTPRequest, name) for name in params}
```
also it might be interesting to provide a default to `getattr`.

Reply:
This is a test, nevermind about the default thing

Reply:
I was not sure whether dict comprehension was already available in 3.7 (and too lazy to check it).

Reply:
Thank you, I sucessfully completed some user tests on Py3.8 playing with some threshold values. So I can control whether my use-case (PDF-generation with JS) works or fails ;-)

In general I have some doubts whether this config is a good idea because even the default may generate troubles. And these errors are difficult to trace. Moreover (D)DOS protection can be considered as a webserver/firewall task and not the task of the app server. Maybe someone has a hint why this complication is needed at all?

Reply:
> I was not sure whether dict comprehension was already available in 3.7 (and too lazy to check it).

According to https://peps.python.org/pep-0274/ it is already in 2.7 and 3.0, so feel free to use it everywhere.